src/test/javascript/test-group-details.js

```diff
@@ -0,0 +1,284 @@
+// diff
+// getAccessType(settings)
+// Group editing
+//      saveInformationForm
+//      saveSettingsForm
+//      deleteGroup
+
+describe("Test group details sidebar functions", function() {
+
+    it("Test diff function for comparing objects", function(done) {
+
+        $.getScript('/src/main/webapp/scripts/group-details-script.js', function() { 
+            var original = {
+                "a": "1",
+                "b": "2",
+                "c": "3",
+            };
+            var diffValues = {
+                "a": "1",
+                "b": "4",
+                "c": "3",
+            };
+            var fewerKeys = {
+                "a": "1",
+                "b": "2",
+            };
+            var empty = {};
+
+            expect(diff(diffValues, original)).toEqual({"b": "4"});
+            expect(diff(original, diffValues)).toEqual({"b": "2"});
+            expect(diff(fewerKeys, original)).toEqual({"c": undefined});
+            expect(diff(original, fewerKeys)).toEqual({});
+            expect(diff(diffValues, fewerKeys)).toEqual({"b": "4"});
+            expect(diff(fewerKeys, diffValues)).toEqual({"b": "2", "c": undefined});
+            expect(diff(original, empty)).toEqual({});
+            expect(diff(empty, original)).toEqual({"a": undefined, "b": undefined, "c": undefined});
+
+            done();
+        });
+    });
+
+    it("Test get access type based on settings parameter", function(done) {
+
+        $.when(
+            $.getScript( '/src/main/webapp/scripts/group-details-script.js' ),
+            $.getScript( '/src/main/webapp/scripts/group-constants.js' ),
+            $.Deferred(function( deferred ){
+                $( deferred.resolve );
+            })
+        ).done(function(){
+            var publicSettings = {
+                "whoCanAdd": "ALL_MANAGERS_CAN_ADD",	
```

Comment:
Have these variables been declared in one of the other script files? if so it should be made a variable scoped for the whole file so it doesn't have to be redeclared here. If its needed for this test can they be created as constants at the top of the file so that it doesn't take up the whole method.